### PR TITLE
Create requests infrastructure

### PR DESCRIPTION
**Pull request by jamohile**
**Background**
Injected content (mainly UI) and background scripts (auth, API) run in separate processes. They can't share state, and can only communicate using [runtime messaging](https://developer.chrome.com/extensions/messaging).

Rather than directly call the messaging API, a **BackgroundRequest** class was designed to easily facilitate communication between the UI and API (background).

**Work Done**
Implemented the necessary infrastructure for BackgroundRequest. This can be extended for each desired request type.

### CODE REVIEW

File: app/babel.config.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+        "@babel/preset-react",
+        "@babel/preset-typescript"
+    ],
+    plugins: [
+        ["@babel/plugin-transform-runtime", {
+            "helpers": false,
+            "regenerator": true
+        }]
+    ]
+}
```
Comment (hananeikhelef) [inline app/babel.config.js:13]: Line  at the end of file ;pp

File: app/tsconfig.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
     "compilerOptions": {
-        "jsx": "react"
-    }
+        "jsx": "react",
+        "esModuleInterop": true
+    },
 }
```
Comment (hananeikhelef) [inline app/tsconfig.json:6]: same here i guess :pp

File: app/babel.config.js (hunk 1 above)
Comment (jamohile) [reply app/babel.config.js]: Fixed!

File: app/tsconfig.json (hunk 2 above)
Comment (jamohile) [reply app/tsconfig.json]: Fixed!

**Pull request merged**